### DISCUSSION TITLE
CS50 Codespace not loading

### DISCUSSION THREAD

**Discussion by w0nton** [community/community#15488] [Codespaces]
Note: I am very new to anything coding-related, so pardon any inability to specify something or unfamiliarity with the subject.
 I am working through the CS50x course and it uses a code space as an IDE. For about a month, it has not loaded when I try to open it. I have tried loading it through the link on the CS50x website, and through GitHub (by going to 'your code spaces' under profile --> All codespaces there are currently 'owned by cs50'). I did at some point, though, get past the 'connecting' loading screen to the 'starting.' I have tried deleting the code space and restarting it in an attempt to change it, yet it didn't seem to work. I'm not sure if this is the wrong approach or not. Currently I have two code spaces listed in my code spaces: solid train, studious guide.

Please let me know what I can do. 

<img width="1133" alt="Screen Shot 2022-04-24 at 1 30 02 PM" src="https://user-images.githubusercontent.com/96895790/164974291-e36b4f84-6e48-4b2e-a9a5-fa4c723acc68.png">
<img width="1334" alt="Screen Shot 2022-04-24 at 1 33 58 PM" src="https://user-images.githubusercontent.com/96895790/164974454-287b8ec0-3bae-4528-8fba-ac34e860ca32.png">

**Comment by shampost** [comment]
Hey @w0nton. I was having the same issue while opening my workspace on the VS Code application. I found that using a prior version (1.5.9) of GitHub Codespaces fixes the issue until CS50 can hammer out the bugs. Let me know if you found anything else that works!

**Reply by w0nton** [reply]
Thanks ! How would I go about doing this and setting this up?

**Reply by shampost** [reply]
Start with downloading the VS Code app... Once you set it up, you can follow Problem Set 1 to access the codespace. Then, in VS Code you should find a panel on the left for extensions. Once there, find the codespace extension, click the settings button and look for "install another version." Let me know if you need any help!

**Reply by w0nton** [reply]
It seems to be working! I didn't exactly follow this, I have had VS Code downloaded for a bit, so I checked if I could download the Codespaces extension- turns out VS Code itself was not updated. Downloaded the newest version, installed the Codespaces extension, saw the cs50 Codespaces - opened it. After some sort of CS50 updates (all of this is still happening in the desktop app) it works! Am able to run the CS50 commands! Thank you

**Reply by Tomi-3-0** [reply]
@w0nton's solution works. The issue seems to be that the latest version of codespaces is only compatible with the latest version of VS Code

**Reply by ni-ghk** [reply]
Good Morning! I`ve recently started the cs50 Python course and I`m having an issue with codespace. Both in VS studio and on the Web version. In the desktop version, it gives me a 404 error. I`ve tried the suggestions above but had no luck :(

**Reply by Tomi-3-0** [reply]
What's the error message

**Reply by ni-ghk** [reply]
These are both errors I get. On the webpage VS: https://share.getcloudapp.com/X6u561Yp And on the actual VS desktop: https://share.getcloudapp.com/nOu1OBg7

**Reply by mkv5173** [reply]
hi i can't login to the codespace via github. It says "BAD GETAWAY Could not GET latest .devcontainer.json"

Can someone help me? I don't know how to fix it. Btw this is on the web

**Reply by mkv5173** [reply]
i tried to create a new codespace but i couldn't find cs50 and i got this error once i selected the main branch

Create New Codespace: List devcontainer definitions request failed with status 400. Correlation ID: E9C0:634B:204BAB6:4285FE7:641E1051

**Reply by ghost** [reply]
Solution: 

I'm sorry to hear that you're experiencing issues with VS Code. This error message typically indicates that there's a problem with the remote server. The first step in troubleshooting this issue is to perform a restart. You can do this by visiting https://cs50.dev/restart. If the problem persists after the restart, you might find it helpful to check out the FAQs regarding CS50 Codespace at https://cs50.readthedocs.io/code/#faqs. These resources should provide you with more information on how to resolve this issue.

**Reply by ghost** [reply]
hi 
i have been having problem with my cs50.dev the sever refused to open 
it have been "sever not found" how can i fix it

**Reply by PAK137** [reply]
![Screenshot (40)](https://github.com/user-attachments/assets/5ce07280-ac2c-474b-8d86-8596a3db4b3b)
Hi I am having issues with code.cs50.io
it doesn't allow me to write in terminal . I have installed VSC in my computer but it doesn't open in that too pls help someone

**Reply by PAK137** [reply]
![Screenshot (40)](https://github.com/user-attachments/assets/60244783-14c1-4b0d-9375-a11749eb2402)
my code.cs50.io is not letting me type in terminal i tried to download VSC in my computer but it did not open in that too help pls...

**Comment by sayhiben** [comment]
As @shampost mentions, this is likely something specific to the course's configuration rather than something broken with GitHub.

**Reply by w0nton** [reply]
Thanks, makes sense.

**Comment by mjsmcd** [comment]
Malon talks to fast and codespace can't take it

**Reply by Euroflower** [reply]
HAHA! I'm glad I'm not the only one who thinks it. He flies through the material. Great instructor though!

**Reply by mjsmcd** [reply]
Yes i listen on youtube and slow down speed of video

**Comment by mjsmcd** [comment]
Trying to use cs50  vscode on my android phone with no computer Seems too dificult Any suggestion

**Reply by ghost** [reply]
buy a computer....suggestion

**Comment by jbalbalosa** [comment]
I just created a new codespace then it worked

**Reply by khushiya26** [reply]
how to create?

**Comment by ByteMe-beep** [comment]
My codespace is stuck at "Setting up your codespace" for the last few days. Any ideas on how can I fix it?

**Reply by gvp100** [reply]
I have the same issue as well

**Reply by click0770** [reply]
My codespace doesn't open at all anymore. It says: "Oh no, something unexpected went wrong!".

**Reply by truncatedjunior** [reply]
same issue i tried deleting my codespace and starting it again but still doesn't seem to work

**Reply by Taha-Alshatiry** [reply]
found any solution??

**Reply by ghost** [reply]
Solution:

I'm sorry to hear that you're experiencing issues with VS Code. This error message typically indicates that there's a problem with the remote server. The first step in troubleshooting this issue is to perform a restart. You can do this by visiting https://cs50.dev/restart. If the problem persists after the restart, you might find it helpful to check out the FAQs regarding CS50 Codespace at https://cs50.readthedocs.io/code/#faqs. These resources should provide you with more information on how to resolve this issue.

**Reply by iftekhar30** [reply]
Try to upgrade your browser.

**Reply by Ash6500** [reply]
In my case, I switched from Google Chrome (has my github logged in) to Mozilla (fresh browser without any pre-occupied data) and logged in to the github account from there. this worked for me fine. You can try it out.

**Reply by AaronMoroney** [reply]
@StealStreet this worked for me

**Comment by gilds** [comment]
Hi

Yes, I went to github with my account and closed all the active codespaces.
Then I was able to open it again.

I hope it helps.

A sexta, 28/07/2023, 11:34, Taha-Alshatiry ***@***.***>
escreveu:

**Reply by liuxiqwe** [reply]
Thanks! My codespace worked now! I think it is the simplest way to resolve this problem.

**Comment by GhostWr1t3r** [comment]
The only solution is to open the codespace in VS CODE 
first go to https://github.com/codespaces
go to cs50  
click to the 3 point and select "open in / with vs code"

**Reply by NanyMina** [reply]
thanks it works now after using your link 
but i want to understand the deffirance between your link and https://cs50.dev/

**Reply by GhostWr1t3r** [reply]
The deffirance is that cs50.dev link maybe is under a lot of requests from the cs50 student so it run in a slow way or it's a cloud browser issues
When u run it from the github codespaces directly on vs code it run in your own code editor in your device so it will be more fastest 

+ https://github.com/codespaces  here you can create / modify codespaces for cs50 or personal uses

**Reply by ArthKulkarni93** [reply]
plz explain clearly!! on that link where are the options to click on like cs50 and open in/with vs code 
I am not able to find those things..... I want to complete the the course ASAP.

**Comment by fish-404** [comment]
I have similar problem, and the codespace seems always timeout. I have tried open in VS code desktop, it doesn't work. Does anyone use offline with docker method successfully?

**Comment by Ju5tu5-224** [comment]
<img width="962" alt="Capture d’écran 2024-09-16 165655" src="https://github.com/user-attachments/assets/daa719b6-396d-4a81-a40b-3d62657210b3">
<img width="970" alt="Capture d’écran 2024-09-20 030522" src="https://github.com/user-attachments/assets/38d521e1-65b2-4a2e-a3cb-4d0abb762654">




please since i started taking the cs50 courses i can't access the cs50 code area please helpe me solve this problem

**Reply by dombrodhun** [reply]
i have the same problem and dont know. most of the time codespace doesnt even start

**Comment by datasec-dot** [comment]
Using the vscode option on my pc worked thanks, but i can't seem to have the intellisense or autocomplete working . am having to type all manually . any help ? thanks in advance

**Comment by ghost** [comment]
hi

**Comment by Mr-coder-z** [comment]
Hi

**Comment by ghost** [comment]
GitHub codespace not open error setting up your codespace why this problem
